File: app/src/main/java/com/fieldservice/app/ui/components/StatusBadge.java
package com.fieldservice.app.ui.components;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.widget.TextView;

import androidx.core.content.ContextCompat;
import androidx.core.graphics.ColorUtils;

import com.fieldservice.app.R;
import com.fieldservice.app.domain.model.TicketStatus;

/** Selo de status do chamado, com rótulo textual (nunca só uma cor). */
public final class StatusBadge {

    private static final int BACKGROUND_ALPHA = (int) (0.16f * 255);

    private StatusBadge() {
    }

    public static String label(TicketStatus status) {
        switch (status) {
            case OPEN:
                return "Aberto";
            case ASSIGNED:
                return "Atribuído";
            case ACCEPTED:
                return "Aceito";
            case TRAVELING:
                return "A caminho";
            case ON_SITE:
                return "No local";
            case IN_PROGRESS:
                return "Em atendimento";
            case WAITING_CONFIRMATION:
                return "Aguardando confirmação";
            case COMPLETED:
                return "Concluído";
            case CANCELLED:
                return "Cancelado";
            default:
                return "";
        }
    }

    private static int colorRes(TicketStatus status) {
        switch (status) {
            case OPEN:
            case ASSIGNED:
                return R.color.status_neutral;
            case ACCEPTED:
            case TRAVELING:
            case ON_SITE:
            case IN_PROGRESS:
            case WAITING_CONFIRMATION:
                return R.color.status_in_progress;
            case COMPLETED:
                return R.color.status_completed;
            case CANCELLED:
                return R.color.status_cancelled;
            default:
                return R.color.color_on_surface;
        }
    }

    public static void bind(TextView view, TicketStatus status) {
        Context context = view.getContext();
        int color = ContextCompat.getColor(context, colorRes(status));

        view.setText(label(status));
        view.setTextColor(color);

        Drawable background = ContextCompat.getDrawable(context, R.drawable.bg_badge);
        if (background != null) {
            background = background.mutate();
            background.setTint(ColorUtils.setAlphaComponent(color, BACKGROUND_ALPHA));
            view.setBackground(background);
        }
    }
}

File: app/src/main/java/com/fieldservice/app/ui/components/StatusBadge.kt
package com.fieldservice.app.ui.components

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.fieldservice.app.domain.model.TicketStatus
import com.fieldservice.app.ui.theme.StatusCancelled
import com.fieldservice.app.ui.theme.StatusCompleted
import com.fieldservice.app.ui.theme.StatusInProgress
import com.fieldservice.app.ui.theme.StatusNeutral

fun TicketStatus.label(): String = when (this) {
    TicketStatus.OPEN -> "Aberto"
    TicketStatus.ASSIGNED -> "Atribuído"
    TicketStatus.ACCEPTED -> "Aceito"
    TicketStatus.TRAVELING -> "A caminho"
    TicketStatus.ON_SITE -> "No local"
    TicketStatus.IN_PROGRESS -> "Em atendimento"
    TicketStatus.WAITING_CONFIRMATION -> "Aguardando confirmação"
    TicketStatus.COMPLETED -> "Concluído"
    TicketStatus.CANCELLED -> "Cancelado"
}

private fun TicketStatus.color(): Color = when (this) {
    TicketStatus.OPEN, TicketStatus.ASSIGNED -> StatusNeutral
    TicketStatus.ACCEPTED, TicketStatus.TRAVELING, TicketStatus.ON_SITE,
    TicketStatus.IN_PROGRESS, TicketStatus.WAITING_CONFIRMATION -> StatusInProgress
    TicketStatus.COMPLETED -> StatusCompleted
    TicketStatus.CANCELLED -> StatusCancelled
}

/** Selo de status do chamado, com rótulo textual (nunca só uma cor). */
@Composable
fun StatusBadge(status: TicketStatus, modifier: Modifier = Modifier) {
    Surface(
        modifier = modifier,
        shape = RoundedCornerShape(6.dp),
        color = status.color().copy(alpha = 0.12f),
        contentColor = status.color()
    ) {
        Text(
            text = status.label(),
            style = MaterialTheme.typography.labelMedium,
            modifier = Modifier.padding(horizontal = 10.dp, vertical = 4.dp)
        )
    }
}
